systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomKeyPairST.java

```diff
@@ -0,0 +1,291 @@
+import io.strimzi.operator.cluster.model.Ca;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.annotations.ParallelNamespaceTest;
+import io.strimzi.systemtest.annotations.ParallelSuite;
+import io.strimzi.systemtest.enums.CustomResourceStatus;
+import io.strimzi.systemtest.kafkaclients.internalClients.KafkaClients;
+import io.strimzi.systemtest.kafkaclients.internalClients.KafkaClientsBuilder;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.security.SystemTestCertHolder;
+import io.strimzi.systemtest.security.SystemTestCertManager;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.RollingUpdateUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.DeploymentUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * Provides test cases to verify custom key-pair (public key + private key) manipulation. For instance:
+ *  1. Replacing `cluster` key-pair (f.e., ca.crt and ca.key) to invoke renewal process
+ *  2. Replacing `clients` key-pair (f.e., user.crt and user.key) to invoke renewal process
+ *
+ * {@link https://strimzi.io/docs/operators/in-development/configuring.html#installing-your-own-ca-certificates-str}
+ * {@link https://strimzi.io/docs/operators/in-development/configuring.html#proc-replacing-your-own-private-keys-str}
+ */
+@ParallelSuite
+public class CustomKeyPairST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(CustomKeyPairST.class);
+
+    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(CustomKeyPairST.class.getSimpleName()).stream().findFirst().get();
+
+    @ParallelNamespaceTest
+    void testReplacingCustomClusterKeyPairToInvokeRenewalProcess(ExtensionContext extensionContext) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+        helperMethod(extensionContext,
+            // ------- public key part
+            (ts, stCertHolder) -> {
+                // 5. Update the Secret for the CA certificate.
```

Comment:
Starting with number 5.)? Can you reference preceding 4. points?

Reply:
Look to the `helperMethod`

Reply:
Yeah, did not get to that at that time :-)